src/main/java/com/faforever/client/replay/LocalReplayVaultController.java

```diff
@@ -50,10 +51,15 @@ public void initialize() {
   protected void onDisplayDetails(ReplayBean replay) {
     JavaFxUtil.assertApplicationThread();
     replayDetailController.setReplay(replay);
+    replayDetailController.getRoot().addEventHandler(DeleteLocalReplayEvent.DELETE_LOCAL_REPLAY_EVENT_TYPE, event -> onLocalReplayDeleted());
     replayDetailController.getRoot().setVisible(true);
     replayDetailController.getRoot().requestFocus();
   }
 
+  private void onLocalReplayDeleted() {
+    onPageChange(searchController.getLastSearchConfig(), false);
+  }
```

Comment:
Will want to convert this to use the google eventBus and so will need to annotate this method with @ Subscribe